test/FastTests/Voron/NextGenPagers/BasicNextGen.cs

```diff
@@ -25,6 +26,114 @@ public BasicNextGen(ITestOutputHelper output) : base(output)
 
     private unsafe static Span<byte> AsSpan(Page p) => new Span<byte>(p.Pointer, Constants.Storage.PageSize);
 
+    [RavenFact(RavenTestCategory.Voron)]
+    public void CannotRecoverWhenDataNotSyncedAndJournalsAreGone()
+    {
+        RequireFileBasedPager();
+
+        Options.ManualFlushing = true;
+        Options.ManualSyncing = true;
+        
+        StartDatabase();
+        StopDatabase(shouldDisposeOptions: true);
+
+        var last = LatestJournalNumber();
+        Env.Options.TryDeleteJournal(last);
+
+        var e = Assert.Throws<InvalidJournalException>(StartDatabase);
+        Assert.Contains("No such journal", e.Message);
+    }
+
+    [RavenFact(RavenTestCategory.Voron)]
+    public void CannotRecoverEvenAfterDataSyncedWhenJournalsAreGone()
+    {
+        RequireFileBasedPager();
+
+        Options.ManualFlushing = true;
+        Options.ManualSyncing = true;
+        
+        StartDatabase();
+
+        using (var tx = Env.WriteTransaction())
+        {
+            tx.LowLevelTransaction.AllocatePage(1);
+            tx.Commit();
+        }
+        
+        Env.FlushLogToDataFile();
+        Env.SyncDataFileImmediately();
+
+        StopDatabase(shouldDisposeOptions: true);
+
+        var last = LatestJournalNumber();
+        Env.Options.TryDeleteJournal(last);
+
+        var e = Assert.Throws<InvalidJournalException>(StartDatabase);
+        Assert.Contains("No such journal", e.Message);
+    }
+
+    [RavenFact(RavenTestCategory.Voron)]
+    public void CannotRecoverAfterTxWhenDataNotSyncedAndJournalsAreGone()
+    {
+        RequireFileBasedPager();
+
+        Options.ManualFlushing = true;
+        Options.ManualSyncing = true;
+        
+        StartDatabase();
+
+        long pageId;
+        using (var tx2 = Env.WriteTransaction())
+        {
+            var allocatePage = tx2.LowLevelTransaction.AllocatePage(1);
+            pageId = allocatePage.PageNumber;
+            tx2.Commit();
+        }
+
+        // we aren't doing flush here to ensure we fail on the start 
+        // Env.FlushLogToDataFile();
+        // Env.SyncDataFileImmediately();
+
+        StopDatabase(shouldDisposeOptions: true);
+
+        var last = LatestJournalNumber();
+        Env.Options.TryDeleteJournal(last);
+
+        var e = Assert.Throws<InvalidJournalException>(StartDatabase);
+        Assert.Contains("No such journal", e.Message);
+    }
+
+    [RavenFact(RavenTestCategory.Voron)]
+    public void CannotRecoverAfterTxWhenDataNotSyncedAndJournalsAreGone2()
+    {
+        RequireFileBasedPager();
+
+        Options.ManualFlushing = true;
+        Options.ManualSyncing = true;
+        
+        StartDatabase();
+
+        long pageId;
+        using (var tx2 = Env.WriteTransaction())
+        {
+            var allocatePage = tx2.LowLevelTransaction.AllocatePage(1);
+            pageId = allocatePage.PageNumber;
+            tx2.Commit();
+        }
+
+        // we aren't doing flush here to ensure we fail on the start 
```

Comment:
Did you mean "sync" instead of "flush"?

Reply:
in general it doesn't matter, when the journal is gone, we can't recover - I adjusted the test